### PR TITLE
Add isFileURL method and use it on all native media upload checks.

### PR DESCRIPTION
**Pull request by SergioEstevao**
## Description

This PR updates the mobile native code instances where the method isURL was being used to check if URL were local URLs (i.e: file urls like: `file:///folder/file.txt`)

~This PR adds a method to check for file URLs, `isFileURL`,  and updates media uploading code to use this method instead of `!isURL(...)`~

This PR replaces the uses of `isURL` to check for file URLs, and uses `getProtocol(url) === ':file'` instead.

## How has this been tested?

Unit tests were added to check the new method.

This can be tested using this mobile GB PR:

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/url/src/is-file-url.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * const isFileURL = isFileURL( 'file:///file.txt' ); // true
+ * ```
+ *
+ * @see https://url.spec.whatwg.org/
+ * @see https://url.spec.whatwg.org/#valid-url-string
+ *
+ * @return {boolean} Whether or not it looks like a file URL.
+ */
+export function isFileURL( url ) {
+	return url.startsWith( 'file:' );
```
Comment (geriux) [inline packages/url/src/is-file-url.js]: I'm still testing, but I wonder if we will have a case where `url` is undefined. In some places we check the url exists before calling this but in some other [places](https://github.com/WordPress/gutenberg/pull/20985/files#diff-f962dfb7b68050764934f0b1cc28d302R83) there aren't any checks. What do you think if we do:

```suggestion
	return url && url.startsWith( 'file:' );
```

Just to be safe =D

Comment (SergioEstevao) [reply packages/url/src/is-file-url.js]: Good catch, I updated the code with that suggestion.

**Review by geriux — APPROVED** [review]
LGTM!  🎉

Tested retry options of Media & Text, Image, Video, and Gallery on both iOS and Android, also ran the tests locally and all ✅

File: packages/url/src/is-file-url.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ *
+ * @example
+ * ```js
+ * const isFileURL = isFileURL( 'file:///file.txt' ); // true
+ * ```
+ *
+ * @return {boolean} Whether or not it looks like a file URL.
+ */
+export function isFileURL( url ) {
+	return url !== undefined && url.startsWith( 'file:' );
```
Comment (aduth) [inline packages/url/src/is-file-url.js]: There's a `getProtocol` function in this module already. Shouldn't we be using it here? Shouldn't it be used _instead_ of adding a new function?

i.e. What's the difference between `isFileURL( url )` and `getProtocol( url ) === 'file:'` ?

**Review by aduth — CHANGES_REQUESTED** [review]

File: packages/url/src/is-file-url.js (hunk 2 above)
Comment (SergioEstevao) [reply packages/url/src/is-file-url.js]: I prefer to have an explicit method for this, but I can use the getProtocol method inside this function. What do you say @aduth ?

Comment (aduth) [reply packages/url/src/is-file-url.js]: Is the preference on the basis that this is a common need? I just wonder how we answer question of which protocols are "deserving" of their own functions. Should we have `isTelephoneURL`? `isSSHURL`? `isMailToURL`? One of the advantages of `getProtocol` is that we don't have to answer that question, nor do we leave any uncertainty to the consuming developers which and why some protocols have specific functions. Is it really enough of a pain point in using `getProtocol`?

Comment (SergioEstevao) [reply packages/url/src/is-file-url.js]: Ok me @aduth talked and decided to refactor the code to use getProtocol to simplify the API and have a more generic approach.

**Comment by SergioEstevao** [comment]
@geriux sorry about this but due to the latest changes can you please review this again, and test with the main apps?

**Review by aduth — APPROVED** [review]
I've not been able to test the impact this might have on the mobile application itself, but at least in addressing my previous concern and in understanding the problem to be that the mobile code previously _relied upon_ a value of `false` from `isURL` as an indication that a URL is a `file:` URL, then the specific code revisions appear to be sensible, and furthermore bring some consistency even within the same file where these conditions are considered (replacing mixed conditions of `url.startsWith( 'file:' )` and `! isURL( url )`.

**Comment by geriux** [comment]
> I've not been able to test the impact this might have on the mobile application itself, but at least in addressing my previous concern and in understanding the problem to be that the mobile code previously _relied upon_ a value of `false` from `isURL` as an indication that a URL is a `file:` URL, then the specific code revisions appear to be sensible, and furthermore bring some consistency even within the same file where these conditions are considered (replacing mixed conditions of `url.startsWith( 'file:' )` and `! isURL( url )`.

Thanks for your feedback and review @aduth, that's why it's always good to have more than one reviewer =)

> sorry about this but due to the latest changes can you please review this again, and test with the main apps?

Don't worry @SergioEstevao ! I'll start testing now 👍

**Comment by geriux** [comment]
Tested it again, the retry options of Media & Text, Image, Video, and Gallery on both iOS and Android  and ✅

LGTM!

**Pull request merged**